### PR TITLE
feat: record burnt_gas in tracing spans

### PR DESCRIPTION
**Pull request by matklad**

### CODE REVIEW

File: runtime/near-vm-runner/src/runner.rs [hunk 1]
```diff
@@ -47,7 +57,10 @@ pub fn run(
             promise_results,
             current_protocol_version,
             cache,
-        )
+        );
+
+        span.record("burnt_gas", &res.outcome().burnt_gas);
```
[the remark this answers is not in the document]
Comment (matklad) [reply runtime/near-vm-runner/src/runner.rs:62]: Good question. "I live in the perfect world of monads" answer would be "yes, we need to record this as gas, and show as tgas, dividing by 10^9 in the presentation layer". But I assume in the world we actually live in it's better to just divide here

[the remark this answers is not in the document]
Comment (matklad) [reply runtime/near-vm-runner/src/runner.rs]: did that in the last commit

File: runtime/near-vm-runner/src/runner.rs [hunk 2]
```diff
@@ -47,7 +57,10 @@ pub fn run(
             promise_results,
             current_protocol_version,
             cache,
-        )
+        );
+
+        span.record("burnt_tera_gas", &(res.outcome().burnt_gas / 10u64.pow(12)));
```
Comment (nagisa) [inline runtime/near-vm-runner/src/runner.rs:62]: Record either a float or gas. Integer Tgas is way too lossy.

File: runtime/near-vm-runner/src/runner.rs (hunk 1 above)
[the remark this answers is not in the document]
Comment (matklad) [reply runtime/near-vm-runner/src/runner.rs:62]: Seems like there are conflicting opinions https://github.com/near/nearcore/pull/6961#discussion_r888995347. For now, let's just record `gas` then, using "this is the simplesht thing we can do" as a tie breaker.

[the remark this answers is not in the document]
Comment (nagisa) [reply runtime/near-vm-runner/src/runner.rs:62]: FWIW jaeger might not have a good way to modify the display, but other distributed systems do allow you to just divide stuff by a constant in sqlish style syntax without a problem.

**Review by nikurt — APPROVED** [review]

**Review by nagisa — APPROVED** [review]

**Pull request merged**